src/screencast/screencast.c

```diff
@@ -271,22 +263,24 @@ static int method_screencast_select_sources(sd_bus_message *msg, void *data,
 		return ret;
 	}
 
-	ret = -1;
+	int output_selection_canceled;
```

Comment:
Not initializing this variable causes building this to fail in Fedora COPR because it detects that it may be uninitialized and -Werror=maybe-unintialized is enabled. It's probably best to set some value here.

Reply:
Good point. Changed. Could you please show me your build configuration. Added -Werror=maybe-unintialized to add_project_arguments and it build regardless. Or does it belong to the default options?

Reply:
I build most of my custom packages using COPR, so you should be able to see the build logs here: https://copr.fedorainfracloud.org/coprs/oranenj/sway/build/1989198/ (don't mind the version number, I just made that up)

I built a fixed version afterwards, setting the initial value to 0, and it produced no complaints.

Reply:
Thanks for the log. Sry, maybe i have to reword my question. Do you know, where and/or how -Werror=maybe-unintialized was added to the meson build process? I'm not that familiar with meson. So it would save me some time, if you could tell me, where we could add it in the meson.build. Tried to add it here https://github.com/columbarius/xdg-desktop-portal-wlr/tree/meson-maybe-unint-test, but this builds without problem on my machine.

Reply:
I notice that I made a typo, it's "maybe-unin**i**tialized", not "maybe-unintialized". But I can't help you with meson either; COPR uses those values as defaults, and I have no idea where they come from :P

Reply:
Luckily i only copied your flag only for the answer :). Damn .... 
Could you please compile my meson test branch on fedora and report, if it works or not, because i can't reproduce it ^^'. It should fail, but doesn't ...

ninja: Entering directory `build'
[0/1] Regenerating build files.
The Meson build system
Version: 0.57.0
Source dir: /volumes/homes/<user>/documents/programming/sway-project/xdg-desktop-portal-wlr
Build dir: /volumes/homes/<user>/documents/programming/sway-project/xdg-desktop-portal-wlr/build
Build type: native build
Project name: xdg-desktop-portal-wlr
Project version: 0.1.0
C compiler for the host machine: cc (gcc 10.2.0 "cc (GCC) 10.2.0")
C linker for the host machine: cc ld.bfd 2.36
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C supports arguments -Wno-missing-braces: YES (cached)
Compiler for C supports arguments -Wno-missing-field-initializers: YES (cached)
Compiler for C supports arguments -Wno-unused-parameter: YES (cached)
Compiler for C supports arguments -Werror=maybe-uninitialized: YES (cached)
Compiler for C supports arguments -D_POSIX_C_SOURCE=200809L: YES (cached)
Library rt found: YES
Dependency libpipewire-0.3 found: YES 0.3.21 (cached)
Dependency wayland-client found: YES 1.19.0 (cached)
Dependency wayland-protocols found: YES 1.20 (cached)
Library iniparser found: YES
Dependency libsystemd found: YES 247 (cached)
Dependency wayland-scanner found: YES 1.19.0 (cached)
Program /usr/bin/wayland-scanner found: YES (/usr/bin/wayland-scanner)
Dependency systemd found: YES 247 (cached)
Configuring xdg-desktop-portal-wlr.service using configuration
Configuring org.freedesktop.impl.portal.desktop.wlr.service using configuration
Build targets in project: 8

Found ninja-1.10.2 at /usr/bin/ninja
[11/11] Linking target xdg-desktop-portal-wlr

Reply:
https://copr.fedorainfracloud.org/coprs/oranenj/sway/build/2006632/ this should be your branch, let's see what happens with it.